### ISSUE TITLE
AccessDeniedPath wrong

### ISSUE THREAD

**Issue by JipingWang** [aspnet/AspNetCore#6669]
### Describe the bug
https://localhost:5001/Account/AccessDenied?ReturnUrl=%2FPrivacy

### To Reproduce
1. New ASP.Net Core 2.2 Web Application
2. Add package
`<PackageReference Include="Microsoft.AspNetCore.Authentication.AzureADB2C.UI" Version="2.2.0" />`
3. Authorize Privacy.cshtml.cs
```
    [Authorize(Roles = "myb2capp-power-user")]
    public class PrivacyModel : PageModel
```
4. configure "AzureAdB2C" in appsettings.json
5. F5 to run
6. click **Privacy** 
7. got 404
> No webpage was found for the web address: https://localhost:5001/Account/AccessDenied?ReturnUrl=%2FPrivacy
> HTTP ERROR 404

### Expected behavior
shoule redirect to 
> https://localhost:5001/AzureADB2C/Account/AccessDenied?ReturnUrl=%2FPrivacy

### Workaround
```
    services.Configure<CookieAuthenticationOptions>(AzureADB2CDefaults.CookieScheme, options => {
            options.AccessDeniedPath = "/AzureADB2C/Account/AccessDenied";
    });
```

### Findings
When using github code "AzureADB2CSample" project for debugging
```
internal class CookieOptionsConfiguration : IConfigureNamedOptions<CookieAuthenticationOptions>{
        private string GetAzureADB2CScheme(string name)
        { //name = "AzureADB2CCookie"
        }
}

```

**Comment by Tratcher** [comment]
Related to https://github.com/aspnet/AspNetCore/issues/4384?

**Comment by Tratcher** [comment]
Hmm, that should have already been configured here:
https://github.com/aspnet/AspNetCore/blob/c7d63649005c4c2863b34794994eef57595e7bb7/src/AADIntegration/src/Microsoft.AspNetCore.Authentication.AzureADB2C.UI/CookieOptionsConfiguration.cs#L31

**Comment by JipingWang** [comment]
@Tratcher , I've fixed it. Could you review the PR? It's just one word change.

**Comment by ankitgupta2k19** [comment]
In my project, the AccessDeniedPath redirects to http instead of https. How do I set it to redirect to https.

**Comment by Tratcher** [comment]
@ankitgupta2k19 that's usually caused by an incorrect proxy configuration. See https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer.

**Comment by muratg** [comment]
cc @Eilon. We'll need to see if this matches patch bar.

**Comment by JipingWang** [comment]
different issues. #6669 fixing a  clear wrong coding variable name, which always causes 404 for a common condition. @muratg

**Comment by Eilon** [comment]
@muratg - do we have a mail thread on this?

**Comment by Eilon** [comment]
@mkArtakMSFT / @javiercn - maybe one of you can start a thread so we can discuss?

**Comment by javiercn** [comment]
@Eilon Sure, let me gather all the details today and I'll start a thread

**Comment by javiercn** [comment]
@Eilon We were deciding where to patch or not. In any case, I'm going to be fixing it on the 3.0 branch. There would be people that could have worked around it and we wanted to avoid breaking those people.

For them to be broken they would have to do `services.Configure<CookieAuthenticationOptions>(AzureADB2CDefaults.CookieScheme, o => o.AccessDeniedPath = "<<Path>>")` before they called
`services.AddAuthentication()
   .AddAzureADB2C();`

If they did it after `services.AddAuthentication().AddAzureADB2C();` then they will continue to override it and won't be broken.

The problem is that they can have gone ahead a put a view/endpoint on that path in order to workaround it, instead of setting the cookie options, so we can't know. I would say however that it should be low risk as it would have required to get into the guts of the code to figure out and not that many people would have done so. But this is all guessing. If you are not confident, we can avoid patching it as there's an easy workaround.

So to summarize:
* Fix it in 3.0 (I'm going to fix a couple of other nuisances along with it all in a batch, will use the contributor commit as  part of the 3.0 work so that is not in vain)
* IMO I wouldn't patch because it has an easy workaround and we could potentially break people. How many of those people, I don't know, but we usually err on the safe side.

Does that sound good?

**Comment by Eilon** [comment]
@javiercn I think that's reasonable. Can you make sure all the relevant issues/PRs/whatever are cleaned up to reflect this?

**Comment by javiercn** [comment]
Yep

**Comment by Eilon** [comment]
@javiercn - any update? Clock is ticking for preview 3.

**Comment by javiercn** [comment]
We said we weren’t patching for this as it could break people who worked around it and has a trivial workaround.

Sent from Outlook
________________________________
